ext/string_metric/levenshtein/iterative_with_two_matrix_rows_ext.c

```diff
@@ -0,0 +1,99 @@
+#include <ruby.h>
+
+#define MIN(a, b, c) ((a) < (b) ? ((a) < (c) ? (a) : (c)) : ((b) < (c) ? (b) : (c)))
+
+#define MALLOC_W(ptr, size) \
+    if ((ptr = malloc(size)) == NULL) \
+        rb_memerror()
+
+VALUE distance_ext(VALUE self, VALUE v_from, VALUE v_to, VALUE v_from_len, VALUE v_to_len,
+                   VALUE v_max_distance, VALUE v_insertion_cost, VALUE v_deletion_cost,
+                   VALUE v_substitution_cost);
+
+VALUE StringMetric = Qnil;
+VALUE Levenshtein = Qnil;
+VALUE IterativeWithTwoMatrixRowsExt = Qnil;
```

Comment:
Global variables of ruby's VALUE type should best be avoided to not hit any strange bugs since those values stay on for the life cycle of a ruby program. (from the moment they `require` the shared library)

In your case you should be safe since you only use them in your Init_ function, but I had to troubleshoot a memory corruption bug few hours ago because i kept a global VALUE results (that i transferred back and forth) and ruby's garbage collector freed it some times.